src/routers/datatypeRouters/_helpers.ts

```diff
@@ -0,0 +1,32 @@
+import { TRPCError } from '@trpc/server'
+
+type paramsFormatType = Record<string, string | number | undefined>
+
+export function preProcessRegionParam (input: paramsFormatType): paramsFormatType {
+  const newInput = input
+  if (input.region !== undefined) {
+    const regex: RegExp = /(chr\w):(\d*)-(\d*)/
+
+    if (regex.test(input.region as string)) {
+      const breakdown = regex.exec(input.region as string)
+      if (breakdown === null || breakdown.length < 4) {
+        throw new TRPCError({
```

Comment:
Can refactor the code so only throw TRPCError once?

Reply:
done